app/routes.py

```diff
@@ -1 +1,268 @@
+from app import db
+import requests
+import os
+
+from app.models.task import Task
+from app.models.goal import Goal
+
+# - - - - - - - - - - - - - - - - - #
+###~~~---- helper function ----~~~###
+# - - - - - - - - - - - - - - - - - #
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"details": f"Invalid data"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message": f"Sorry, {cls.__name__} {model_id} does not exist"}, 404))
+
+    return model
+
+# - - - - - - - - - - - - - - - - - #
+###~~~--- task model routes ---~~~###
+# - - - - - - - - - - - - - - - - - #
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@task_bp.route("", methods=['POST'])
+
+# define a route for creating a task
+def create_task():
+    request_body = request.get_json()
+
+    try:
+        new_task = Task(
+            title = request_body["title"],
+            description = request_body["description"],
+        )
+    except:
+        abort(make_response({"details": f"Invalid data"}, 400))
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+# define a route for getting all tasks
+@task_bp.route("", methods=["GET"])
+def read_all_tasks():
+    # querries
+    title_query = request.args.get("title")
+    description_query = request.args.get("description")
+    completed_at_query = request.args.get("completed_at")
+    sort_query = request.args.get("sort")
+
+    if title_query:
+        tasks = Task.query.filter_by(title=title_query)
+    
+    elif description_query:
+        tasks = Task.query.filter_by(description=description_query)
+    
+    elif completed_at_query:
+        tasks = Task.query.filter_by(completed_at=completed_at_query)
+    
+    elif sort_query:
+        if sort_query == 'asc':
+            tasks = Task.query.order_by(Task.title).all()
+        elif sort_query == 'desc':
+            tasks = Task.query.order_by(Task.title.desc()).all()
+    
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+
+    return jsonify(tasks_response), 200
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+
+    if task.goal_id:
+        return {
+        "task": {
+            "id": task.task_id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": True if task.completed_at else False
+        }
+    }, 200
+
+    return {"task": task.to_dict()}, 200
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
Right now, if a user sends a request without the keys title or description your server would crash. There's a couple of ways to handle this, you could call the `validate_request` function before you access the keys in `request_body` or you could implement a `try/except` block.